.github/workflows/quick.yaml

```diff
@@ -40,13 +40,21 @@ jobs:
 
   functionality-tests:
 
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
 
     steps:
+      # XXX: A hack to fix Ubuntu runners that were created in April 2025.
+      - name: Fix /etc/hosts
+        run: |
+          cat /etc/hosts
+          cp -p /etc/hosts /tmp/etc-hosts.bak
+          sudo sed --in-place -E 's/^(-e.*)/# \1/' /etc/hosts
+          diff -u /tmp/etc-hosts.bak /etc/hosts || true
+
```

Comment:
I am happy to remove this out-of-scope hack from this PR (with or without submitting it in a dedicated PR). Just let me know your preference.

```
Image: ubuntu-24.04
Version: 20250427.1.0
```

Without this hack, this PR is likely to fail our CI functionality tests like all the other recent PRs that are unlucky enough to run on Ubuntu runners built in April 2025. I was hoping that /etc/hosts was not broken on Ubuntu v24.04 runners, but those runners have the same problem -- it looks like GitHub was using the same broken process/automation to generate both Ubuntu LTS runners :-(.

We should still upgrade to test master on the latest LTS version -- this PR would remain valid even if this hack is removed.